feature(mobile): restore scan utxo

jorbuedo opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Introduced advanced QR code scanning with improved camera permission handling and user feedback.
  * Added UTxO consolidation and management screens to enhance wallet performance.
  * Added haptic feedback for various actions.
  * Introduced a centralized navigation system for smoother app flow.
  * Added new hooks for bold text formatting and next-tick scheduling.

* **Improvements**
  * Refined localization with new and updated messages for scan, send, and transaction features.
  * Enhanced error handling and UI feedback across wallet setup, send, and receive flows.
  * Improved component structure, theming, and import organization throughout the app.
  * Updated dependency versions and added new package dependencies.

* **Bug Fixes**
  * Fixed typos and improved consistency in string keys and import paths.
  * Corrected navigation and import path issues for various features.

* **Documentation**
  * Updated development guides with new best practices and Android testing instructions.

* **Chores**
  * Disabled currency name translation in settings.
  * Cleaned up unused or duplicate code and reorganized file exports.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jorbuedo committed:
feat: restore Send, Receive, and UTXO features with comprehensive navigation

- Add Send screens to TxHistoryNavigator (send-start-tx, send-list-amounts-to-send, send-edit-amount, send-select-token-from-list, send-submitted-tx, send-failed-tx)
- Add Receive screens to TxHistoryNavigator (receive-single, receive-specific-amount, receive-multiple)
- Add UTXO screens to TxHistoryNavigator (utxo-list, utxo-consolidation)
- Add comprehensive navigation functions to useWalletNavigation for Send, Receive, and UTXO features
- Update ActionsBanner to use centralized navigation functions
- Add missing string messages for Send, Receive, and UTXO features
- Fix import paths for navigation functions across multiple files
- Restore complete Send and Receive transaction flows
- Enable UTXO list and consolidation functionality

This commit restores the core wallet functionality for sending and receiving transactions, as well as UTXO management features.

jorbuedo committed:
fix: correct import paths for KeyboardAvoidingView and navigation modules

- Fix KeyboardAvoidingView import path from ~/ui/KeyboardAvoidingView to ~/ui/KeyboardAvoidingView/KeyboardAvoidingView
- Fix navigation import paths from ~/kernel/navigation to ~/kernel/navigation/navigation
- Resolves Metro bundler import resolution errors
- Affects RequestSpecificAmountScreen, SwapSettings, ReviewSwap, CreateExchangeOrderScreen, Dashboard, FailedTxScreen, InfraestructureIssueScreen, and ReviewTxScreen

jorbuedo committed:
feat: create index files for navigation and KeyboardAvoidingView modules

- Add apps/mobile/src/kernel/navigation/index.ts to re-export all navigation types and functions
- Add apps/mobile/src/ui/KeyboardAvoidingView/index.ts to re-export KeyboardAvoidingView component
- Revert specific import path fixes to use the cleaner index file approach
- This resolves all import path issues across the codebase by providing proper module entry points
- Fixes Metro bundler import resolution errors for both navigation and KeyboardAvoidingView modules

jorbuedo committed:
fix: resolve missing SubmittedTxIcon component imports

- Fix SubmittedTxIcon import in Send/ShowSubmittedTxScreen/SubmittedTxScreen.tsx
- Replace with SuccessfulTxIcon from ~/ui/SuccessfulTxIcon/SuccessfulTxIcon
- Add missing imports in ReviewTx/ShowSubmittedTxScreen/SubmittedTxScreen.tsx
- Add useStrings, atoms, and SuccessfulTxIcon imports
- Resolves Metro bundler import resolution errors for missing components

jorbuedo committed:
Fix IconProps import and memoMaxLenght import paths

jorbuedo committed:
doc

jorbuedo committed:
feat: implement camera scanner functionality

- Add CameraCodeScanner component with visual mask
- Implement haptics feedback for scan success/error
- Add scan-start and scan-show-camera-permission-denied routes
- Fix react-native-permissions dependency issues
- Use expo-camera built-in barcode scanning
- Add proper error handling and user feedback
- Enable QR code scanning in Send flow

jorbuedo committed:
feat: implement modern QR scanner with Expo Camera

- Replace legacy Camera with modern CameraView
- Use useCameraPermissions for proper permission handling
- Implement QR code scanning with onBarcodeScanned
- Add proper error handling and haptic feedback
- Fix query dehydration issues with address validation
- Use atoms for styling throughout
- Integrate with existing scan action system
- Add navigation from tx history and send flow

jorbuedo committed:
Merge develop into feature/scan-camera-implementation

jorbuedo committed:
Fix various issues in Send flow and UTXO list

- Fix BigInt literals in SelectTokenFromListScreen for ES2020 compatibility
- Fix import path for limitOfSecondaryAmountsPerTx constant
- Fix typo in selectAssetTitle string key
- Add missing useTheme imports in NoSpendableAmount and EmptySearchResult components
- Fix style issues using proper atoms instead of invalid textAlign
- Fix noAssetsAddedYet string to be a function with proper translation key
- Fix EditAmountScreen null check to prevent 'quantity' of undefined error
- Update useStatusBar hook with proper error handling for expo-system-ui

banklesss committed:
feature(mobile): integrate receive

banklesss committed:
Update TokenChart.tsx

Signed-off-by: banklesss <105349292+banklesss@users.noreply.github.com>

banklesss committed:
Update TokenChart.tsx

Signed-off-by: banklesss <105349292+banklesss@users.noreply.github.com>

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
+import {atoms as a, useTheme} from '@yoroi/theme'
+import {CameraView, useCameraPermissions} from 'expo-camera'
+import * as Haptics from 'expo-haptics'
+import React, {useCallback, useRef, useState} from 'react'
```
``` suggestion
import * as React from 'react'
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
-    (event: BarCodeScannerResult) => {
-      scannerRef.current?.stopScanning()
-      const {data: codeContent} = event
+  const handleBarCodeScanned = useCallback(
```
```suggestion
  const handleBarCodeScanned = React.useCallback(
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
+    insideFeature: insideFeature as any,
+  })
+  const [permission, requestPermission] = useCameraPermissions()
+  const [scanned, setScanned] = useState(false)
+  const cameraRef = useRef<CameraView>(null)
```
```suggestion
  const [scanned, setScanned] = React.useState(false)
  const cameraRef = React.useRef<CameraView>(null)
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
+  })
+  const {insideFeature} = scanParamsSchema.parse(params)
+  const triggerScanAction = useTriggerScanAction({
+    insideFeature: insideFeature as any,
```
should be parsed at this point, (any lint rule should be complaining 👀 )

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
-      onCameraPermissionDenied={navigateTo.showCameraPermissionDenied}
-      withMask
-    />
+  const handleScanAgain = useCallback(() => {
```
```suggestion
  const handleScanAgain = React.useCallback(() => {
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
+  }, [])
+
+  useFocusEffect(
+    useCallback(() => {
```
```suggestion
    React.useCallback(() => {
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
 
 export const ScanCodeScreen = () => {
-  const navigateTo = useNavigateTo()
+  const {palette: p} = useTheme()
```
```suggestion
  const {palette: p, atoms: ta} = useTheme()
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,195 @@
+          a.flex_1,
+          a.justify_center,
+          a.align_center,
+          {backgroundColor: p.bg_color_max},
```
```suggestion
          ta.bg_color_max,
```

jorbuedo committed:
fix(scan): address PR review feedback from stackchain

- Use React.useCallback, React.useState, React.useRef for consistency
- Use atoms as a for layout styling and ta for background colors
- Fix insideFeature type to be more specific
- Improve code formatting and consistency

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+          style={[
+            a.pt_lg,
+            a.p_md,
+            {backgroundColor: p.el_primary_medium},
```
```suggestion
            ta.el_primary_medium,
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+            a.rounded_md,
+          ]}
+        >
+          <Text style={[a.body_1_lg_regular, {color: p.text_primary_max}]}>
```
```suggestion
          <Text style={[a.body_1_lg_regular, ta.text_primary_max]}>
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+          ]}
+        >
+          <Text style={[a.body_1_lg_regular, {color: p.text_primary_max}]}>
+            Grant Permission
```
i18n

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+        style={[a.flex_1, a.justify_center, a.align_center, ta.bg_color_max]}
+      >
+        <Text style={[a.body_1_lg_regular, {color: p.gray_max}]}>
+          We need your permission to show the camera
```
i18n

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+        style={[a.flex_1, a.justify_center, a.align_center, ta.bg_color_max]}
+      >
+        <Text style={[a.body_1_lg_regular, {color: p.gray_max}]}>
+          Requesting camera permission...
```
i18n

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -5,71 +1,195 @@
+                {color: p.gray_max},
+              ]}
+            >
+              QR Code Scanned Successfully
```
it was marked as solved, but it needs the translations

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+                style={[
+                  a.flex_1,
+                  a.p_md,
+                  {backgroundColor: p.el_primary_medium},
```
```suggestion
                  ta.el_primary_medium,
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+                onPress={() => navigateToTxHistory()}
+              >
+                <Text
+                  style={[a.body_1_lg_regular, {color: p.text_primary_max}]}
```
```suggestion
                  style={[a.body_1_lg_regular, ta.text_primary_max]}
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+                style={[
+                  a.flex_1,
+                  a.p_md,
+                  {backgroundColor: p.el_gray_medium},
```
```suggestion
                  ta.el_gray_medium,
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+                onPress={handleScanAgain}
+              >
+                <Text
+                  style={[a.body_1_lg_regular, {color: p.text_primary_max}]}
```
```suggestion
                  style={[a.body_1_lg_regular, ta.text_primary_max]}
```

stackchain commented on apps/mobile/src/features/Scan/useCases/ScanCodeScreen.tsx:
```diff
@@ -1,75 +1,178 @@
+                <Text
+                  style={[a.body_1_lg_regular, {color: p.text_primary_max}]}
+                >
+                  Scan Again
```
i18n

stackchain commented on apps/mobile/src/features/Send/common/useSendAddress.tsx:
```diff
@@ -19,12 +19,7 @@ export const useSendAddress = () => {
-      enabled: false,
-    },
-  )
+  } = useValidateAddress({address, chainId})
```
this is no longer async, maybe can become that a regular function doesnt need to be a hoook anymore.

stackchain commented on apps/mobile/src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx:
```diff
@@ -217,7 +217,7 @@ const ListSpendables = ({
       )}
       bounces={false}
-      contentContainerStyle={[a.px_lg]}
+      contentContainerStyle={{paddingHorizontal: 16}}
```
is it the Style complaining about the types? my guess is they are not properly typing the Props.
```suggestion
      contentContainerStyle={{paddingHorizontal: 16}}
```

jorbuedo replied:
Doesn't like the atoms inside an array, in this case since there's only 1 we can use it directly. Doing that now. I'll change them everywhere.

stackchain commented on apps/mobile/src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx:
```diff
@@ -360,12 +361,7 @@ const NoSpendableAmount = ({text}: {text: string}) => {
-          {color: p.gray_max, flex: 1, textAlign: 'center'},
-        ]}
-      >
+      <Text style={[a.heading_3_medium, {color: p.gray_max, flex: 1}]}>
```
I think all p.gray_max needs to be moved to some ta.text_ something, need to check with design
```suggestion
      <Text style={[a.heading_3_medium, {color: p.gray_max}, a.flex_1]}>
```

jorbuedo replied:
There's an element one maybe?

stackchain replied:
there are some text_...

stackchain commented on apps/mobile/src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx:
```diff
@@ -384,12 +381,7 @@ const EmptySearchResult = () => {
-          {color: p.gray_max, flex: 1, textAlign: 'center'},
-        ]}
-      >
+      <Text style={[a.heading_3_medium, {color: p.gray_max, flex: 1}]}>
```
```suggestion
      <Text style={[a.heading_3_medium, {color: p.gray_max}, a.flex_1]}>
```

stackchain commented on apps/mobile/src/features/Send/useCases/ListAmountsToSend/AddToken/Show/MaxAmountsPerTx.tsx:
```diff
@@ -2,24 +2,22 @@ import {useTheme} from '@yoroi/theme'
-      >
+      <Text style={[{color: p.gray_max, fontSize: 14, lineHeight: 20}]}>
         <Text
-          style={theme.atoms.body_2_md_medium}
-        >{`${limitOfSecondaryAmountsPerTx} ${strings.send.assets.toLocaleLowerCase()} `}</Text>
+          style={{fontSize: 14, lineHeight: 20, fontWeight: '500'}}
+        >{`${limitOfSecondaryAmountsPerTx} `}</Text>
```
!error inline styles facilitate the integration between mobile / desktop styling and also collocate the code.

stackchain commented on apps/mobile/src/features/Send/useCases/ListAmountsToSend/EditAmount/EditAmountScreen.tsx:
```diff
@@ -51,15 +51,24 @@ export const EditAmountScreen = () => {
 
   const amount = targets[selectedTargetIndex].entry.amounts[selectedTokenId]
+
+  if (!amount) {
+    // If amount is not found, navigate back to the token selection screen
+    React.useEffect(() => {
+      navigateTo.selectedTokens()
+    }, [navigateTo])
+    return null
+  }
```
this issue is correct, theoretically it wont happen.

stackchain commented on apps/mobile/src/features/Send/useCases/ListAmountsToSend/AddToken/Show/MaxAmountsPerTx.tsx:
```diff
@@ -2,24 +2,22 @@ import {useTheme} from '@yoroi/theme'
-      <Text
-        style={[theme.atoms.body_2_md_regular, {color: theme.color.gray_max}]}
-      >
+      <Text style={[{color: p.gray_max, fontSize: 14, lineHeight: 20}]}>
```
it was expanded from the atoms, i think it can be reverted

stackchain commented on apps/mobile/src/features/Send/useCases/ListAmountsToSend/ListAmountsToSendScreen.tsx:
```diff
@@ -137,7 +136,7 @@ export const ListAmountsToSendScreen = () => {
     >
       <AmountsList
-        data={Object.values(amounts)}
+        data={Object.values(amounts) as Portfolio.Token.Amount[]}
```
it should infer automatically is it complaining?

jorbuedo replied:
Removing null | undefined I think, but no it's not complaining, removing the as.

stackchain commented on apps/mobile/src/features/Send/useCases/StartMultiTokenTx/InputReceiver/InputReceiver.tsx:
```diff
@@ -2,11 +2,11 @@ import {useTheme} from '@yoroi/theme'
 import React from 'react'
```
```suggestion
import * as React from 'react'
```

jorbuedo replied:
Changed everywhere

stackchain commented on apps/mobile/src/features/Transactions/TxHistoryNavigator.tsx:
```diff
@@ -22,39 +42,234 @@ export const TxHistoryNavigator = () => {
+    return resolverApiMaker({
+      apiConfig: {
+        [Resolver.NameServer.Unstoppable]: {
+          apiKey: '', // Empty for now, can be configured later
```
should be reading from the env

stackchain commented on apps/mobile/src/features/Transactions/useCases/UtxoConsolidation/UtxoConsolidation/UtxoConsolidation.tsx:
```diff
@@ -45,7 +45,7 @@ export const UtxoConsolidation = () => {
               <Icon.Info size={20} color={p.el_gray_max} />
 
               <Text style={[a.body_1_lg_medium, {color: p.el_gray_medium}]}>
```
```suggestion
              <Text style={[a.body_1_lg_medium, ta.el_gray_medium]}>
```

stackchain commented on apps/mobile/src/features/Transactions/useCases/UtxoList/UtxoItem.tsx:
```diff
@@ -15,16 +15,16 @@ export const UtxoItem = ({item}: {item: UtxoList[number]['utxos'][number]}) => {
   const {palette: p} = useTheme()
```
```suggestion
  const {palette: p, atoms: ta} = useTheme()
```

stackchain commented on apps/mobile/src/features/Transactions/useCases/UtxoList/UtxoItem.tsx:
```diff
@@ -15,16 +15,16 @@ export const UtxoItem = ({item}: {item: UtxoList[number]['utxos'][number]}) => {
   const utxoId = `${item.txHash}#${item.txIndex}`
 
-  if (tokenInfos === undefined)
+  if (isLoading || !tokenInfos || tokenIds.length === 0)
```
this will cause an 0 list to stay on loading forever

stackchain commented on apps/mobile/src/features/Transactions/useCases/UtxoList/UtxoItem.tsx:
```diff
@@ -15,16 +15,16 @@ export const UtxoItem = ({item}: {item: UtxoList[number]['utxos'][number]}) => {
-      tokenIds: Object.keys(item.balance) as Portfolio.Token.Id[],
-      sourceId: 'UtxoList',
-    })
+  const tokenIds = Object.keys(item.balance) as Portfolio.Token.Id[]
```
don't see any advantage in this update and it also falls back to casting

jorbuedo replied:
Object.keys converts them to strings

stackchain commented on apps/mobile/src/features/Transactions/useCases/UtxoList/useUtxoList.ts:
```diff
@@ -26,7 +26,6 @@ export const useUtxoList = () => {
   useWalletEvent(wallet, 'utxos', () => queryClient.invalidateQueries(queryKey))
 
   const query = useQuery({
```
if before was relying on suspense for the loading state

```suggestion
  const query = useSuspenseQuery({
```

jorbuedo replied:
It wasn't

stackchain commented on apps/mobile/src/kernel/haptics/feedback.ts:
I think this can be deleted since the impl before was consuming straight from the Haptics

stackchain commented on apps/mobile/src/kernel/i18n/index.ts:
oww boy, no index files.

stackchain commented on apps/mobile/src/features/WalletManager/hooks/index.ts:
no index files

stackchain commented on apps/mobile/src/kernel/i18n/messages/global.ts:
```diff
@@ -2,7 +2,7 @@ import {freeze} from 'immer'
 
 import {ThemeName} from '@yoroi/theme'
-import {supportedCurrencies} from '~/features/Settings/useCases/changeAppSettings/Currency/CurrencyContext'
+// import {supportedCurrencies} from '~/features/Settings/useCases/changeAppSettings/Currency/CurrencyContext'
```
i think is safe to uncomment

stackchain commented on apps/mobile/src/kernel/i18n/messages/global.ts:
```diff
@@ -634,49 +634,49 @@ export const actionMessages = freeze(
-  }),
-  true,
-)
+// export const currencyNames: Readonly<
```
ibid idem

stackchain commented on apps/mobile/src/kernel/i18n/messages/send.ts:
```diff
@@ -198,24 +207,28 @@ export const sendMessages = defineMessages({
   },
   helperAddressErrorWrongBlockchain: {
-    id: 'send.helper.addressError.wrongBlockchain',
+    id: 'components.send.sendscreen.addressInputLabel',
```
it it correct?

stackchain commented on apps/mobile/src/kernel/i18n/messages/send.ts:
```diff
@@ -198,24 +207,28 @@ export const sendMessages = defineMessages({
   },
   helperAddressErrorWrongNetwork: {
-    id: 'claim.apiError.invalidRequest',
+    id: 'components.send.sendscreen.addressInputLabel',
```
-

stackchain commented on apps/mobile/src/kernel/i18n/useStrings.ts:
```diff
@@ -538,10 +540,10 @@ export const useStrings = () => {
-          translatedName: (symbol: Portfolio.Currency.Symbol) =>
-            f(currencyNames[symbol]),
-        },
+        // currencySettings: {
```
-

stackchain commented on apps/mobile/src/kernel/logger/index.ts:
to be deleted

stackchain commented on apps/mobile/src/kernel/navigation/index.ts:
to be deleted

stackchain commented on apps/mobile/src/kernel/query/QueryProvider.tsx:
```diff
@@ -35,7 +35,13 @@ export function QueryProvider({children}: React.PropsWithChildren) {
-          shouldDehydrateQuery: () => true,
+          shouldDehydrateQuery: (query) => {
+            // Don't dehydrate queries that explicitly opt out
+            if (query.meta?.shouldDehydrate === false) {
+              return false
+            }
+            return true
```
```suggestion
            return query.meta?.shouldDehydrate !== false
```

stackchain replied:
i think it worth simplify the logic, is exacly the same

stackchain commented on apps/mobile/src/ui/Analytics/Analytics.tsx:
```diff
@@ -235,7 +235,11 @@ const CommonContent = ({
             <Text style={style}>{icon}</Text>
 
-            <Text style={{...a.body_1_lg_regular}}>{strings.ui[key]}</Text>
+            <Text style={{...a.body_1_lg_regular}}>
```
ta not a. (im guessing)
```suggestion
            <Text style={[a.body_1_lg_regular]}>
```

stackchain commented on apps/mobile/src/ui/Analytics/Analytics.tsx:
```diff
@@ -235,7 +235,11 @@ const CommonContent = ({
-            <Text style={{...a.body_1_lg_regular}}>{strings.ui[key]}</Text>
+            <Text style={{...a.body_1_lg_regular}}>
+              {key === 'private' || key === 'noip' || key === 'nosell'
+                ? strings.ui[key](bold)
```
(bold)?

jorbuedo replied:
Changed everywhere, bold is the function now coming from useBold hook that changes the <b> to Text with bold atom

stackchain commented on apps/mobile/src/ui/Button/index.ts:
to be deleted

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+import {atoms as a, useTheme} from '@yoroi/theme'
+import {CameraView, useCameraPermissions} from 'expo-camera'
+import React, {useCallback, useState} from 'react'
```
```suggestion
import * as React from 'react'
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+  const [permission, requestPermission] = useCameraPermissions()
+  const [scanned, setScanned] = useState(false)
+
+  const handleBarCodeScanned = useCallback(
```
```suggestion
  const handleBarCodeScanned = React.useCallback(
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+  withMask,
+  maskText,
+}: CameraCodeScannerProps) => {
+  const {palette: p} = useTheme()
```
```suggestion
  const {palette: p, atoms: ta} = useTheme()
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+  )
+
+  if (!permission)
+    return <View style={[a.flex_1, {backgroundColor: p.bg_color_max}]} />
```
No message?
```suggestion
    return <View style={[a.flex_1, ta.bg_color_max]} />
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+        ]}
+      >
+        <Text style={[a.body_1_lg_regular, {color: p.gray_max}]}>
+          We need your permission to show the camera
```
i18n

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+          style={[
+            a.pt_lg,
+            a.p_md,
+            {backgroundColor: p.el_primary_medium},
```
```suggestion
            ta.el_primary_medium,
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+            a.rounded_md,
+          ]}
+        >
+          <Text style={[a.body_1_lg_regular, {color: p.text_primary_max}]}>
```
```suggestion
          <Text style={[a.body_1_lg_regular, ta.text_primary_max]}>
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+  }
+
+  return (
+    <View style={[a.flex_1, {backgroundColor: p.bg_color_max}]}>
```
```suggestion
    <View style={[a.flex_1, ta.bg_color_max]}>
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+          ]}
+        >
+          <Text style={[a.body_1_lg_regular, {color: p.text_primary_max}]}>
+            Grant Permission
```
i18n

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/CameraCodeScanner.tsx:
```diff
@@ -0,0 +1,107 @@
+              style={[
+                a.pt_lg,
+                a.body_1_lg_regular,
+                {color: p.text_primary_max, textAlign: 'center'},
```
```suggestion
                ta.text_primary_max,
                a.align_center,
```

stackchain commented on apps/mobile/src/ui/CameraCodeScanner/index.ts:
to be deleted

stackchain commented on apps/mobile/src/ui/Icon/index.ts:
to be deleted

stackchain commented on apps/mobile/src/ui/KeyboardAvoidingView/index.ts:
to be deleted

jorbuedo replied:
Delete what, the whole index for Icons? Why? I think it's pretty good to have Icon.Analytics instead of confusing an Analytics icon with an Analytics component.

stackchain commented on apps/mobile/src/ui/PairedBalance/index.ts:
to be deleted

stackchain commented on apps/mobile/src/ui/ShareDetailsCard/ShareDetailsCard.tsx:
```diff
@@ -1,14 +1,14 @@
 import {atoms as a, useTheme} from '@yoroi/theme'
+import {LinearGradient} from 'expo-linear-gradient'
 import React from 'react'
```
```suggestion
import * as React from 'react'
```

stackchain commented on apps/mobile/src/ui/ShareQRCodeCard/ShareQRCodeCard.tsx:
```diff
@@ -7,14 +8,11 @@ import {
 import QRCode from 'react-native-qrcode-svg'
-import Share from 'react-native-share'
-import ViewShot, {captureRef} from 'react-native-view-shot'
+// import ViewShot, {captureRef} from 'react-native-view-shot'
```
```suggestion
// TODO: REVISIT implement a similar package (maybe expo has a better one)
// import ViewShot, {captureRef} from 'react-native-view-shot'
```

stackchain commented on apps/mobile/src/ui/Space/index.ts:
to be deleted

stackchain commented on apps/mobile/src/ui/TextInput/index.ts:
to be deleted

stackchain commented on apps/mobile/src/ui/TokenAmountItem/TokenAmountItem.tsx:
```diff
@@ -131,7 +128,7 @@ export const TokenAmountItem = ({
         {variant !== 'swap' && (
           <PairedBalance
-            textStyle={[{textAlign: 'right'}]}
+            textStyle={{textAlign: 'right'}}
```
```suggestion
            textStyle={[a.align_right]}
```

stackchain commented on apps/mobile/src/ui/TokenAmountItem/TokenAmountItem.tsx:
```diff
@@ -146,7 +143,7 @@ export const TokenAmountItem = ({
 
                 <PairedBalance
-                  textStyle={[{textAlign: 'right'}]}
+                  textStyle={{textAlign: 'right'}}
```
```suggestion
                  textStyle={[a.align_right]}
```

stackchain commented on apps/mobile/src/ui/TokenAmountItem/index.ts:
to be deleted

stackchain commented on apps/mobile/src/wallets/utils/index.ts:
to be deleted

stackchain commented on apps/mobile/temp_icon_index.ts:
to be deleted

jorbuedo committed:
Merge branch 'develop' into feature/restore-send-receive-utxo-camera-navigation

jorbuedo committed:
feat: restore send/receive/utxo/camera navigation and fix styling issues

- Updated development guide with atomic design system best practices
- Fixed SaveNanoXScreen.tsx: corrected imports, error handling, and styling
- Refactored useBold hook for centralized bold text formatting
- Removed unnecessary index.ts barrel export files
- Added new i18n strings for scan feature
- Fixed hardcoded strings in scan components
- Updated styling to use themed atoms (ta) instead of direct palette
- Fixed navigation issues in various components
- Improved error handling with proper errorMessages usage
- Enhanced component consistency and maintainability

jorbuedo committed:
Merge branch 'develop' into feature/receive-integration-2

jorbuedo committed:
feat: resolve conflicts with feature/receive-integration-2

- Combined resolver functionality with receive provider
- Integrated receive functionality into ActionsBanner
- Fixed import conflicts in ShareDetailsCard and ShareQRCodeCard
- Maintained our navigation structure while incorporating their receive features
- Added sharing functionality to ShareQRCodeCard

jorbuedo committed:
feat: finalize conflict resolution with receive integration

jorbuedo committed:
Merge branch 'feature/conflict-resolution-receive-integration' into feature/restore-send-receive-utxo-camera-navigation

jorbuedo committed:
rm temp index

jorbuedo committed:
refactor: replace inline textAlign styles with atomic design system

- Replace {textAlign: 'right'} with a.text_right in TokenAmountItem
- Replace {textAlign: 'right'} with a.text_right in AmountCard
- Follow atomic design system guidelines for consistent styling
- Improve maintainability and type safety

jorbuedo committed:
refactor: improve React imports and atom usage

- Fix React imports in CameraCodeScanner to use * as React pattern
- Replace useCallback and useState with React.useCallback and React.useState
- Fix atom spreading in Analytics component
- Replace inline color and textAlign styles with atomic design system atoms
- Use ta.text_primary_medium and a.text_center instead of inline styles

jorbuedo committed:
fix: use environment variable for Unstoppable API key

- Import unstoppableApiKey from ~/kernel/constants
- Replace hardcoded empty string with environment variable
- Follows rn71 branch pattern for API key configuration

jorbuedo committed:
refactor: apply new ta rules to UtxoConsolidation component

- Replace inline color styles with themed atoms (ta)
- Update {color: p.gray_900} to ta.text_primary_max
- Update {color: p.el_gray_medium} to ta.text_gray_medium
- Update Icon.Info color to use ta.el_gray_max.color
- Follow atomic design system guidelines for theme-aware styling

jorbuedo committed:
refactor: apply new ta rules to all UtxoList components

- UtxoItem.tsx: Replace {color: p.el_gray_max} with ta.text_gray_max
- UtxoItem.tsx: Replace ActivityIndicator color with ta.el_gray_medium.color
- WarningSingleAddress.tsx: Replace {color: p.gray_max} with ta.text_gray_max
- UtxoAddressGroup.tsx: Replace {color: p.el_gray_max} with ta.text_gray_max
- UtxoAddressGroup.tsx: Replace {color: p.el_gray_medium} with ta.text_gray_medium
- UtxoAddressGroup.tsx: Replace {color: p.primary_500} with ta.text_primary_medium
- Follow atomic design system guidelines for theme-aware styling

jorbuedo committed:
fix: revert ActivityIndicator to use direct palette access

- ActivityIndicator color should use direct palette access (p.el_gray_medium)
- UI elements like loading indicators should maintain consistent color
- Keep themed atoms (ta) for text content that should adapt to theme

jorbuedo committed:
refactor: apply new ta rules across entire codebase

- Replace inline color styles with themed atoms (ta) throughout the app
- Update {color: p.*} patterns to use appropriate ta.* atoms
- Follow atomic design system guidelines for theme-aware styling
- Maintain direct palette access for UI elements that need consistent behavior
- Improve code formatting and consistency

jorbuedo committed:
refactor: remove unused haptics feedback wrapper

- Delete apps/mobile/src/kernel/haptics/feedback.ts (unused)
- Remove empty haptics directory
- ScanCodeScreen.tsx already uses expo-haptics directly
- Clean up unused code to reduce bundle size and complexity

jorbuedo committed:
refactor: uncomment supportedCurrencies and currencyNames, fix send address validation, and update error message IDs

stackchain commented on apps/mobile/src/kernel/navigation/navigation.tsx:
this file was deleted in the base branch, something went off during the merging, i think is just imports.

stackchain reviewed (approved):
the navigation file was rebased incorrectly, yet is a easy fix, bringing it since it is a ton of updates.
